src/Microsoft.Health.Dicom.Blob/Features/Export/AzureBlobExportSink.cs

```diff
@@ -63,22 +64,37 @@ public async Task<bool> CopyAsync(ReadResult value, CancellationToken cancellati
             return false;
         }
 
-        using Stream sourceStream = await _source.GetStreamingFileAsync(value.Instance.VersionedInstanceIdentifier.Version, value.Instance.VersionedInstanceIdentifier.Partition, value.Instance.InstanceProperties.fileProperties, cancellationToken);
-        BlobClient destBlob = _dest.GetBlobClient(_output.GetFilePath(value.Instance.VersionedInstanceIdentifier));
-
+        InstanceMetadata i = value.Instance;
         try
         {
-            await destBlob.UploadAsync(sourceStream, new BlobUploadOptions { TransferOptions = _blobOptions.Upload }, cancellationToken);
+            using Stream sourceStream = await _source.GetStreamingFileAsync(i.VersionedInstanceIdentifier.Version,
+                i.VersionedInstanceIdentifier.Partition, i.InstanceProperties.fileProperties, cancellationToken);
+            BlobClient destBlob = _dest.GetBlobClient(_output.GetFilePath(i.VersionedInstanceIdentifier));
+            await destBlob.UploadAsync(sourceStream, new BlobUploadOptions { TransferOptions = _blobOptions.Upload },
+                cancellationToken);
             return true;
         }
+        catch (DataStoreException ex) when (ex.IsExternal)
```

Comment:
Why call this out specifically? `DataStoreRequestFailedException` doesn't derive from `DataStoreException`, does it?